src/Components/WebAssembly/WebAssembly/src/Http/WebAssemblyHttpRequestMessageExtensions.cs

```diff
@@ -0,0 +1,169 @@
+                BrowserRequestCache.OnlyIfCached => "only-if-cached",
+                _ => throw new InvalidOperationException($"Unsupported enum value {requestCache}.")
+            };
+
+            return SetBrowserFetchOption(requestMessage, "cache", stringOption);
+        }
+
+        /// <summary>
+        /// Configures a value for the 'mode' option for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="requestMode">The <see cref="BrowserRequestMode"/>.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>\
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/Request/mode
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserRequestMode(this HttpRequestMessage requestMessage, BrowserRequestMode requestMode)
+        {
+            if (requestMessage is null)
+            {
+                throw new ArgumentNullException(nameof(requestMessage));
+            }
+
+            var stringOption = requestMode switch
+            {
+                BrowserRequestMode.SameOrigin => "same-origin",
+                BrowserRequestMode.NoCors => "no-cors",
+                BrowserRequestMode.Cors => "cors",
+                BrowserRequestMode.Navigate => "navigate",
+                _ => throw new InvalidOperationException($"Unsupported enum value {requestMode}.")
+            };
+
+            return SetBrowserFetchOption(requestMessage, "mode", stringOption);
+        }
+
+        /// <summary>
+        /// Configures a value for the 'integrity' option for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="integrity">The subresource integrity descriptor.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/Request/integrity
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserIntegrity(this HttpRequestMessage requestMessage, string integrity)
+            => SetBrowserFetchOption(requestMessage, "integrity", integrity);
+
+        /// <summary>
+        /// Configures a value for the HTTP request.
+        /// </summary>
+        /// <param name="requestMessage">The <see cref="HttpRequestMessage"/>.</param>
+        /// <param name="name">The name of the HTTP fetch option.</param>
+        /// <param name="value">The value, which must be JSON-serializable.</param>
+        /// <returns>The <see cref="HttpRequestMessage"/>.</returns>
+        /// <remarks>
+        /// See https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch
+        /// </remarks>
+        public static HttpRequestMessage SetBrowserFetchOption(this HttpRequestMessage requestMessage, string name, object value)
```

Comment:
```suggestion
        public static HttpRequestMessage SetBrowserRequestOption(this HttpRequestMessage requestMessage, string name, object value)
```

Similar reason to above.

Also, in JavaScript the interface defining the options is called `RequestInit`, so it sounds right to use the term "request" rather than "fetch" from a JS perspective too.